### PR TITLE
Add return type annotations in grapher - part 2

### PR DESCRIPTION
**Pull request by danyx23**
This adds the grapher folders barCharts, bodyDiv, captionedChart, controls and a few leftovers in chart

### CODE REVIEW

File: grapher/controls/Controls.tsx [hunk 1]
```diff
@@ -213,48 +213,48 @@ export interface FooterControlsManager extends ShareMenuManager {
...
+                                        ? " active"
+                                        : "")
+                                }
+                            >
+                                <a
+                                    onClick={():
+                                        | GrapherTabOption.chart
+                                        | GrapherTabOption.map
+                                        | GrapherTabOption.sources
+                                        | GrapherTabOption.table =>
```
Comment (MarcelGerber) [inline grapher/controls/Controls.tsx]: I was very confused about this for a few seconds, and don't really like this extreme case.

File: grapher/controls/entityPicker/EntityPicker.tsx [hunk 2]
```diff
@@ -114,32 +119,39 @@ export class EntityPicker extends React.Component<{
...
-    @computed private get availableEntitiesForCurrentView() {
+    @computed private get availableEntitiesForCurrentView(): Set<string> {
         if (!this.manager.requiredColumnSlugs?.length || !this.grapherTable)
             return this.selection.availableEntityNameSet
         return this.grapherTable.entitiesWith(this.manager.requiredColumnSlugs)
     }
 
     @computed
-    private get entitiesWithMetricValue(): EntityOptionWithMetricValue[] {
+    private get entitiesWithMetricValue() {
```
Comment (MarcelGerber) [inline grapher/controls/entityPicker/EntityPicker.tsx:146]: You _removed_ the return type here?

**Review by MarcelGerber — COMMENTED** [review]
Thank you. This looks good overall, and I've pushed a minor commit for the few cases where it was a bit wonky ;)

In general, I'm unsure about the value explicit return types provide us in anonymous functions, like in `map`, `forEach`, etc.

Maybe the [`allowExpressions`](https://github.com/typescript-eslint/typescript-eslint/blob/cae4f4a0f33f8c954b1670d0abcfc8edd6193a06/packages/eslint-plugin/docs/rules/explicit-function-return-type.md#allowexpressions) rule might be something for us? (Haven't tried it!)

Ideally (at least in my mind), we would have explicit return types for every function that is somehow assigned a name (no matter in which way), and return types would be optional for everything else. Don't know if that's possible, though.

File: grapher/controls/entityPicker/EntityPicker.tsx (hunk 2 above)
Comment (danyx23) [reply grapher/controls/entityPicker/EntityPicker.tsx:146]: Thanks for catching this! I tried to automate some of the annotations which occasionally removed an annotation and I missed this one.

**Comment by danyx23** [comment]
Thanks a lot Marcel for catching these cases! 

I agree that on some arrow functions the return type annotations don't give us too much. I looked through this PR again and I noticed that several of them are not even necessary (i.e. the warnings we activate don't even flag them) and I introduced some of them with the attempt to automate some of the return type annotations. I have the redundant annotations as they are mostly visual clutter and don't get us much as you say.

I'll adapt my routine and check more carefully the cases of any automated annotation if I'll still use it at all

**Comment by danielgavrilov** [comment]
Whoops sorry, I intended that commit for the part 1 branch, got them mixed up!

But leaving it here is probably fine since they're getting merged?

**Comment by danyx23** [comment]
Yes should be totally fine. I'll add a note in slack in the developers channel

**Pull request merged**